include/zephyr/bluetooth/audio/bap.h

```diff
@@ -296,7 +287,7 @@ struct bt_bap_scan_delegator_recv_state {
 	uint8_t num_subgroups;
 
 	/** Subgroup specific information */
-	struct bt_bap_scan_delegator_subgroup subgroups[BT_BAP_SCAN_DELEGATOR_MAX_SUBGROUPS];
+	struct bt_bap_base_subgroup subgroups[CONFIG_BT_BAP_BASE_MAX_SUBGROUPS];
```

Comment:
Ditto: `CONFIG_BT_BAP_BASE_MAX_SUBGROUPS` is too generic (and we have in fact just removed a similar Kconfig).